go/enclave/evm/evm_facade.go

```diff
@@ -196,7 +199,7 @@ func executeTransaction(
 
 		// Create a new context to be used in the EVM environment
 		blockContext := gethcore.NewEVMBlockContext(header, bc, author)
-		vmenv := vm.NewEVM(blockContext, vm.TxContext{BlobHashes: tx.Tx.BlobHashes(), GasPrice: header.BaseFee}, statedb, config, cfg)
+		vmenv = vm.NewEVM(blockContext, vm.TxContext{BlobHashes: tx.Tx.BlobHashes(), GasPrice: header.BaseFee}, statedb, config, cfg)
```

Comment:
Not really sure why this `applyTx` nested function exists, makes this quite hard to follow. But seems like vmenv will always be created or we'd have returned with error before you use it outside the function so guess it works.

Reply:
tbh, I don't remember either. I think @StefanIliev545 might have worked in this area as well.
At some point we were depending on geth internal functions, which eventually became public